From 0b9699c1602cde643e9ee4914a335aa7a887556d Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Wed, 22 May 2019 13:25:02 +0200
Subject: [PATCH 137/345] MIPS: Octeon: Cleanup obsolete CrashKernel memory
 init in octeon/setup.c

Make it compile with KDUMP and KEXEC features enabled.

Signed-off-by: Corey Minyard <cminyard@mvista.com>
Signed-off-by: David Daney <david.daney@cavium.com>
---
 arch/mips/cavium-octeon/setup.c | 14 ++------------
 1 file changed, 2 insertions(+), 12 deletions(-)

diff --git a/arch/mips/cavium-octeon/setup.c b/arch/mips/cavium-octeon/setup.c
index a9425f5dd165..0cadea8fa154 100644
--- a/arch/mips/cavium-octeon/setup.c
+++ b/arch/mips/cavium-octeon/setup.c
@@ -1103,17 +1103,6 @@ void __init plat_mem_setup(void)
 	if (mem_alloc_size > max_memory)
 		mem_alloc_size = max_memory;
 
-/* Crashkernel ignores bootmem list. It relies on mem=X@Y option */
-#ifdef CONFIG_CRASH_DUMP
-	add_memory_region(reserve_low_mem, max_memory, BOOT_MEM_RAM);
-	total += max_memory;
-#else
-#ifdef CONFIG_KEXEC
-	if (crashk_size > 0) {
-		add_memory_region(crashk_base, crashk_size, BOOT_MEM_RAM);
-		crashk_end = crashk_base + crashk_size;
-	}
-#endif
 	cvmx_bootmem_lock();
 	while ((boot_mem_map.nr_map < BOOT_MEM_MAP_MAX)
 		&& (total < max_memory)) {
@@ -1159,8 +1148,9 @@ void __init plat_mem_setup(void)
 		}
 	}
 	cvmx_bootmem_unlock();
-#endif /* CONFIG_CRASH_DUMP */
+
 mem_alloc_done:
+
 #ifdef CONFIG_CAVIUM_RESERVE32
 	/*
 	 * Now that we've allocated the kernel memory it is safe to
-- 
2.25.1

